src/ipa/rpi/controller/rpi/clock_recovery.h

```diff
@@ -0,0 +1,43 @@
+#include <chrono>
+#include <list>
+using namespace std::chrono_literals;
+
+namespace RPiController {
+
+class ClockRecovery
+{
+public:	
+	std::list<int64_t> error_values;
+	std::list<int64_t> client_sequence;
+	unsigned int syncPeriod = 30;
+	unsigned int listSize = 100;
+	void updating_values(std::chrono::microseconds delta);
+	std::chrono::microseconds trending_error(std::chrono::microseconds LastWallClock, std::chrono::microseconds ClientWallClock, std::chrono::microseconds lastPayloadFrameDuration, unsigned int sequence);
+	int64_t modeled_wall_clock(int64_t WallClock, int64_t KernelWallClock, unsigned int sequence);
+	void clear();
```

Comment:
Might be worth double-checking if nothing above could also be private. For example, `listSize` feels a bit like an internal implementation detail.